shell/key-bindings.bash

```diff
@@ -53,21 +53,16 @@ __fzf_cd__() {
 
 __fzf_history__() (
   local line
-  shopt -u nocaseglob nocasematch
   line=$(
-    HISTTIMEFORMAT= builtin history |
-    FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} $FZF_DEFAULT_OPTS --tac --sync -n2..,.. --tiebreak=index --bind=ctrl-r:toggle-sort $FZF_CTRL_R_OPTS +m" $(__fzfcmd) |
-    command grep '^ *[0-9]') &&
-    if [[ $- =~ H ]]; then
-      sed 's/^ *\([0-9]*\)\** .*/!\1/' <<< "$line"
-    else
-      sed 's/^ *\([0-9]*\)\** *//' <<< "$line"
-    fi
+    builtin fc -lnr -2147483648 |
+      perl -p -l0 -e 'BEGIN { getc; $/ = "\n\t" } s/^[ *]//; $_ = $ENV{HISTCMD} - $. . "\t$_"' |
```

Comment:
Any reason you manually put index numbers instead of using `fc` without `-n`?

Reply:
With `-n` you don't need to parse the index numbers with a regular expression. You can split the history list with the input record separator, which to me seemed clean. Also it makes the item format independent of `fc`'s output, in case something different is preferred.